sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnInvokers.java

```diff
@@ -126,6 +135,38 @@ private DoFnInvokers() {}
     return constructor;
   }
 
+  /** Default implementation of {@link DoFn.SplitRestriction}, for delegation by bytebuddy. */
+  public static class DefaultSplitRestriction {
+    /** Doesn't split the restriction. */
+    @SuppressWarnings("unused")
+    public static <InputT, RestrictionT> List<RestrictionT> invokeSplitRestriction(
+        InputT element, RestrictionT restriction) {
+      return Collections.singletonList(restriction);
+    }
+  }
+
+  /** Default implementation of {@link DoFn.SplitRestriction}, for delegation by bytebuddy. */
+  public static class DefaultRestrictionCoder {
+    private static final CoderRegistry CODER_REGISTRY = new CoderRegistry();
+
+    static {
+      CODER_REGISTRY.registerStandardCoders();
```

Comment:
This may not be generally safe. If the user has registered a restriction coder, this won't be picked up here, for instance.